### PR TITLE
Drops within radius

### PR DESCRIPTION
**Pull request by langjoh**
Maybe some specs are still missing, but the radius response works

### CODE REVIEW

File: app/controllers/api/drops_controller.rb [hunk 1]
```diff
@@ -2,6 +2,12 @@ class Api::DropsController < ApplicationController
   respond_to :json
 
   def index
-    @drops = Drop.all
+    if params[:latitude] && params[:longitude]
+      radius = params[:radius].presence || 10
+      places_within_radius = Place.near([params[:latitude], params[:longitude]], radius, units:  :km)
+      @drops = Drop.where(place_id: places_within_radius.map(&:id))
```
Comment (berlintam) [inline app/controllers/api/drops_controller.rb]: you can also use a super awesome method here called pluck - which it a but faster than `map`.
ex: `@drops = Drop.where(place_id: places_within_radius.pluck(:id))`

File: lib/distance.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# class Distance
+#   include Geocoder::Calculations
+#
+#
+# end
```
Comment (berlintam) [inline lib/distance.rb]: is this file needed? can you delete it?

File: spec/lib/distance_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# require 'Geocoder'
+# require 'rails_helper'
+# require './lib/distance.rb'
+#
+# RSpec.describe Distance::Geocoder::Calculations do
+#   it 'returns the distance between two points' do
+#     distance = distance_between([52.538255, 13.396255], [52.537016, 13.394861], options = {:units => :km})
+#     expect(distance).to include(0.244)
+#   end
+# end
```
Comment (berlintam) [inline spec/lib/distance_spec.rb]: can you delete this too? is it needed?

File: spec/requests/api/drops_controller_spec.rb [hunk 4]
```diff
@@ -27,9 +29,46 @@
...
         expect(parsed_response[0]["place"].keys).to eq(["id", "name", "latitude", "longitude"])
       end
+    end
+
+
+    describe "GET /api/drops" do
+
+      context 'with latitude and longitude given' do # will always set the radio to 10km
+        let!(:drop_in_berlin) { FactoryGirl.create :drop, :drop_in_rheinsberger}
+        let!(:drop_in_sydney) { FactoryGirl.create :drop, :drop_in_sydney}
```
Comment (berlintam) [inline spec/requests/api/drops_controller_spec.rb]: there's a short cut for Factory Girl, instead you can do:
`let!(:drop_in_berlin) { create :drop, :drop_in_rheinsberger }`

I _think_ you can do this because we included this line in our rails helper, which gives us access to alternate (shorter) factory girl methods. BUT i'm not 100% sure about that so don't quote me on it!
https://github.com/sounddrop/sounddrop/blob/master/spec/rails_helper.rb#L55

File: spec/requests/api/drops_controller_spec.rb [hunk 5]
```diff
@@ -27,9 +29,46 @@
...
+        get "/api/drops?latitude=52.537016&longitude=13.394861&radius=0.5" # 1 Km away from SoundCloud
+        parsed_response = ActiveSupport::JSON.decode(response.body)
+        expect(parsed_response.length).to eq(1)
+      end
+
+  #     it "" do
+  #       get "/api/drops?latitude=4.701647&longitude=-74.041916" # Somewhere in Bogotá
+  #       parsed_response = ActiveSupport::JSON.decode(response.body)
+  #       expect(parsed_response.length).to eq(0)
+ #         end
```
Comment (berlintam) [inline spec/requests/api/drops_controller_spec.rb]: you can delete this too

File: app/controllers/api/drops_controller.rb [hunk 6]
```diff
@@ -2,6 +2,12 @@ class Api::DropsController < ApplicationController
   respond_to :json
 
   def index
-    @drops = Drop.all
+    if params[:latitude] && params[:longitude]
+      radius = params[:radius].presence || 10
```
Comment (berlintam) [inline app/controllers/api/drops_controller.rb]: I would suggest creating a constant for the number `10`. 
I guess it would go in the drops model. So you could call it: `DEFAULT_RADIUS = 10`. Just so that we make sure we always use the same radius as default everywhere throughout the app & remain consistent. 

Let me know if you need help w/ this!

File: app/models/place.rb [hunk 7]
```diff
@@ -6,5 +6,6 @@ class Place < ActiveRecord::Base
   validates :location, presence: true
 
   geocoded_by :location
-  after_validation :geocode
+
+  after_validation :geocode, unless: :geocoded?
```
Comment (berlintam) [inline app/models/place.rb:10]: oooOoOoOoh smart.

**Comment by berlintam** [comment]
great job! One thing to keep in mind is to delete commented out stuff. You can always access it gain via the github history if you need it again so it's not going anywhere. Leaving it in the code just makes things a bit messy, so best to get rid of it. Remember, delete all the things!

woot woot

File: lib/distance.rb (hunk 2 above)
Comment (aydamacink) [inline lib/distance.rb]: yes, team delete!

File: spec/requests/api/drops_controller_spec.rb (hunk 5 above)
Comment (aydamacink) [inline spec/requests/api/drops_controller_spec.rb]: done!

File: spec/lib/distance_spec.rb (hunk 3 above)
Comment (aydamacink) [inline spec/lib/distance_spec.rb]: done!

File: lib/distance.rb (hunk 2 above)
Comment (aydamacink) [inline lib/distance.rb]: done!

File: app/controllers/api/drops_controller.rb (hunk 1 above)
Comment (aydamacink) [inline app/controllers/api/drops_controller.rb:11]: We are not using pluck - which is  faster than map, because it created problems with sql see https://github.com/alexreisner/geocoder/issues/166 and the README : "Because the SELECT clause is modified, anything else that modifies the SELECT clause may produce strange results, for example:

```
using the pluck method (selects only a single column)"
```

File: app/controllers/api/drops_controller.rb (hunk 6 above)
Comment (langjoh) [inline app/controllers/api/drops_controller.rb]: done :) We added the constant in the controllerfor now.

File: app/controllers/api/drops_controller.rb [hunk 8]
```diff
@@ -1,11 +1,12 @@
 # We use .map instead of .pluck, as .pluck modifies the SELECT clause and produces strange results.
 
 class Api::DropsController < ApplicationController
+  DEFAULT_RADIUS = 10
   respond_to :json
 
   def index
     if params[:latitude] && params[:longitude]
-      radius = params[:radius].presence || 10
+      radius = params[:radius].presence || DEFAULT_RADIUS
```
Comment (berlintam) [inline app/controllers/api/drops_controller.rb:9]: 👍

File: spec/requests/api/drops_controller_spec.rb (hunk 5 above)
Comment (langjoh) [inline spec/requests/api/drops_controller_spec.rb]: done :)

File: spec/requests/api/drops_controller_spec.rb (hunk 4 above)
Comment (aydamacink) [inline spec/requests/api/drops_controller_spec.rb]: done!

File: spec/requests/api/drops_controller_spec.rb (hunk 5 above)
Comment (aydamacink) [inline spec/requests/api/drops_controller_spec.rb]: done!

**Comment by berlintam** [comment]
fantastic
